src/test/scala/SchedulerTest.scala

```diff
@@ -0,0 +1,75 @@
+import java.util.concurrent.{ExecutorService, Executors, PriorityBlockingQueue}
+
+import org.scalatest.concurrent.Eventually
+import org.scalatest.{BeforeAndAfter, BeforeAndAfterAll, Matchers, WordSpec}
+
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+
+class SchedulerTest extends WordSpec with Matchers with Eventually with BeforeAndAfterAll with BeforeAndAfter {
+
+  object TimeOrder extends Ordering[ScheduledTask] {
+    override def compare(x: ScheduledTask, y: ScheduledTask): Int = x.timestamp compare y.timestamp
+  }
+
+  val NUM_OF_THREAD = 5
+  val QUEUE_SIZE = 100
+  val SECOND_IN_MILLI = 1000
+
+  val ex: ExecutorService = Executors.newFixedThreadPool(NUM_OF_THREAD)
+  implicit val ec: ExecutionContext = ExecutionContext.fromExecutorService(ex)
+
+  /**
+    * scalatest BeforeAndAfterAll provides setUpModule/tearDownModule
+    * this runs before/after this file execution
+    */
+
+  override def beforeAll(): Unit = {
+
+  }
+
+  override def afterAll(): Unit = {
+    ex.shutdown()
```

Comment:
`ex.shutdown()` stops assigning further tasks to threads in the thread pool, but wait for currently executing task in a thread to finish. this will prevent `run-away` threads.